### PR TITLE
Deprecate use of roles in favor of recipes

### PR DESCRIPTION
**Pull request by ramereth**
This pulls all the magic we do in roles currently into the recipes. This greatly simplifies many things with this cookbook.

### CODE REVIEW

**Comment by ramereth** [comment]
@osuosl-cookbooks/chefs please review. This will require a related chef-repo change.

File: recipes/controller.rb [hunk 1]
```diff
@@ -16,14 +16,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+vnc_bind_int = node['osl-openstack']['vnc_bind_interface']['controller']
+node.default['openstack']['endpoints']['compute-vnc-bind']['bind_interface'] =
```
Comment (jordane) [inline recipes/controller.rb]: While we're at it, please make this `= Chef::DelayedEvaluator.new { node['osl-openstack']['vnc_bind_interface']['controller'] }`

**Comment by ramereth** [comment]
@osuosl-cookbooks/chefs please re-review

File: recipes/controller.rb [hunk 2]
```diff
@@ -16,14 +16,42 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+node.default['openstack']['endpoints']['compute-vnc-bind']['bind_interface'] =
+  Chef::DelayedEvaluator.new do
+    node['osl-openstack']['vnc_bind_interface']['controller']
+  end
```
Comment (ramereth) [inline recipes/controller.rb]: This introduces a new error :sob: 

```
       NoMethodError
       -------------
       undefined method `[]' for nil:NilClass
       
       Cookbook Trace:
       ---------------
         /tmp/kitchen/cache/cookbooks/openstack-common/libraries/network.rb:29:in `address_for'
         /tmp/kitchen/cache/cookbooks/openstack-common/libraries/endpoints.rb:101:in `address'
         /tmp/kitchen/cache/cookbooks/openstack-common/libraries/uri.rb:40:in `uri_from_hash'
         /tmp/kitchen/cache/cookbooks/openstack-common/libraries/endpoints.rb:32:in `endpoint'
         /tmp/kitchen/cache/cookbooks/openstack-compute/recipes/nova-common.rb:98:in `from_file'
         /tmp/kitchen/cache/cookbooks/openstack-compute/recipes/nova-setup.rb:25:in `from_file'
         /tmp/kitchen/cache/cookbooks/osl-openstack/recipes/controller.rb:43:in `from_file'
       
       Relevant File Content:
       ----------------------
       /tmp/kitchen/cache/cookbooks/openstack-common/libraries/network.rb:
       
        22:  module ::Openstack # rubocop:disable Documentation
        23:    # return the IPv4 (default) address of the given interface.
        24:    #
        25:    # @param [String] interface The interface to query.
        26:    # @param [String] family The protocol family to use.
        27:    # @return [String] The address.
        28:    def address_for(interface, family = node['openstack']['endpoints']['family'])
        29>>     interface_node = node['network']['interfaces'][interface]['addresses']
        30:      interface_node.select do |address, data|
        31:        return address if data['family'] == family
        32:      end
        33:    end
        34:  end
        35:  
```

Comment (jordane) [inline recipes/controller.rb]: Ugh, I was worried about this. Not everything works with `Chef::DelayedEvaluator`, unfortunately. I guess just revert it for now.

**Comment by ramereth** [comment]
@osuosl-cookbooks/chefs please re-re-review

**Comment by jordane** [comment]
+1

**Pull request merged**